packages/lodestar/test/unit/network/encoders/request.test.ts

```diff
@@ -1,124 +1,70 @@
-      all
-    );
+chai.use(chaiAsPromised);
 
-    expect(requests.length).to.be.equal(1);
-    expect(config.types.Uint64.equals(requests[0].body, BigInt(0))).to.be.true;
-  });
+describe("network / encoders", () => {
+  describe("request encoder", () => {
+    for (const encoding of [ReqRespEncoding.SSZ, ReqRespEncoding.SSZ_SNAPPY]) {
+      it(`simulate request basic - ${encoding}`, async function () {
+        const requestBody = BigInt(0);
+        const returnedRequest = await simulateRequest(Method.Ping, encoding, requestBody);
 
-  it("should work - container request - ssz", async function () {
-    const status = createStatus();
-    const requests = await pipe(
-      [status],
-      eth2RequestEncode(config, logger, Method.Status, ReqRespEncoding.SSZ),
-      eth2RequestDecode(config, logger, Method.Status, ReqRespEncoding.SSZ),
-      all
-    );
-
-    expect(requests.length).to.be.equal(1);
-    expect(config.types.Status.equals(requests[0].body, status)).to.be.true;
-  });
+        expect(config.types.Uint64.equals(returnedRequest.body as typeof requestBody, requestBody)).to.be.true;
+      });
 
-  it("should work - container request - ssz", async function () {
-    const status = createStatus();
-    const requests = await pipe(
-      [status],
-      eth2RequestEncode(config, logger, Method.Status, ReqRespEncoding.SSZ_SNAPPY),
-      eth2RequestDecode(config, logger, Method.Status, ReqRespEncoding.SSZ_SNAPPY),
-      all
-    );
-
-    expect(requests.length).to.be.equal(1);
-    expect(config.types.Status.equals(requests[0].body, status)).to.be.true;
-  });
+      it(`simulate request container - ${encoding}`, async function () {
+        const requestBody = createStatus();
+        const returnedRequest = await simulateRequest(Method.Status, encoding, requestBody);
 
-  it("should work - multiple request - ssz", async function () {
-    const requests = await pipe(
-      [BigInt(1), BigInt(2)],
-      eth2RequestEncode(config, logger, Method.Ping, ReqRespEncoding.SSZ),
-      eth2RequestDecode(config, logger, Method.Ping, ReqRespEncoding.SSZ),
-      all
-    );
+        expect(config.types.Status.equals(returnedRequest.body as typeof requestBody, requestBody)).to.be.true;
+      });
 
-    expect(requests.length).to.be.equal(1);
-    expect(config.types.Uint64.equals(requests[0].body, BigInt(1))).to.be.true;
-  });
+      it(`simulate request bad body - ${encoding}`, async function () {
+        const requestBody = BigInt(0);
 
-  it("should work - multiple request - ssz_snappy", async function () {
-    const requests = await pipe(
-      [BigInt(1), BigInt(2)],
-      eth2RequestEncode(config, logger, Method.Ping, ReqRespEncoding.SSZ_SNAPPY),
-      eth2RequestDecode(config, logger, Method.Ping, ReqRespEncoding.SSZ_SNAPPY),
-      all
-    );
+        await expect(simulateRequest(Method.Status, encoding, requestBody)).to.be.rejectedWith(
+          "Cannot convert undefined or null to object"
+        );
+      });
```

Comment:
Can you add test for request without body?

Reply:
streamRequestBodyTo is not called for a request without a body, there's an if in sendRequestStream